app/components/UpcomingEvent/index.js

```diff
@@ -0,0 +1,26 @@
+import React, { PropTypes } from 'react';
+
+const UpcomingEvent = ({ event, index }) => (
+  <article className="UpcomingEvent" key={index}>
```

Comment:
Apply the key to the `UpcomingEvent` and only pass in event. This will stop UpcomingEvent from having knowledge about the content it is used in (i.e. it's one in a collection), and means we have no chance of property of undefined error later when we index into `event`